packages/jest-emotion/test/matchers.test.js

```diff
@@ -120,4 +120,132 @@ describe('toHaveStyleRule', () => {
     const resultPass = toHaveStyleRule(tree, 'color', 'red')
     expect(resultPass.message()).toMatchSnapshot()
   })
+
+  it('matches styles on the focus, hover targets', () => {
+    const localDivStyle = emotion.css`
+      color: white;
+        &:hover {
+          color: yellow;
+        }
+         &:focus {
+          color: black;
+        }
+    `
+    const tree = renderer
+      .create(
+        <div className={localDivStyle}>
+          <svg className={svgStyle} />
+        </div>
+      )
+      .toJSON()
+
+    expect(tree).toHaveStyleRule('color', 'yellow', { target: 'hover' })
+    expect(tree).toHaveStyleRule('color', 'black', { target: 'focus' })
+    expect(tree).toHaveStyleRule('color', 'white')
+  })
+
+  it('matches styles on the nested component or html element', () => {
+    const Svg = styled('svg')`
+      width: 100%;
```

Comment:
If I were to set a color here, `fill: blue;`, which is overridden later to `fill: green;` and the test checks for 'blue' will the tests still pass even though the end visual result is a `green` fill?

Reply:
The test check for `fill: blue` will fail.

Reply:
The test check for `fill: blue`. The `fill` would be `green`.

Reply:
Could you add a test for that?

Reply:
Yes, I've added it. https://github.com/emotion-js/emotion/pull/1110/files#diff-f37132e84fc7f1a6f14f4ae5bd1d96cdR150

Reply:
I don't think that tests it, that looks like it's declaring the fill property once.

Reply:
@mitchellhamilton lets have a chat in slack, I have sent you a message there